Sources/MongoDriver.swift

```diff
@@ -2,176 +2,267 @@ import Foundation
+            throw Error.unsupported
         }
         
-        return try database[query.entity].insert(document)
+        return (limit.count, limit.offset)
     }
-
-    private func select<T: Entity>(_ query: Fluent.Query<T>) throws -> Cursor<Document> {
-        let cursor: Cursor<Document>
-
-        let aqt = try query.makeAQT()
-        let mkq = MKQuery(aqt: aqt)
-        let sortDocument: Document?
+    
+    /// Transforms Fluent Query data into a Document
+    ///
+    /// Used for Create/Update operations
+    private func makeDocument(_ data: [RawOr<String>: RawOr<Node>]) throws -> Document {
+        var document = Document()
         
-        if !query.sorts.isEmpty {
-            let elements = query.sorts.map { ($0.field, $0.direction == .ascending ? Value.int32(1) : Value.int32(-1)) }
-            sortDocument = Document(dictionaryElements: elements)
-        } else {
-            sortDocument = nil
+        for (key, value) in data {
+            guard let key = key.wrapped, let value = value.wrapped else {
+                throw Error.invalidData
+            }
+            
+            document[key] = value
         }
         
-        if let limit = query.limit {
-            cursor = try database[query.entity].find(matching: mkq,
-                                                     sortedBy: sortDocument,
-                                                     skipping: Int32(limit.offset),
-                                                     limitedTo: Int32(limit.count))
-        } else {
-            cursor = try database[query.entity].find(matching: mkq,
-                                                     sortedBy: sortDocument)
+        if let key = document.type(at: "_id"), case ElementType.nullValue = key {
+            document["_id"] = nil
         }
-
-        return cursor
+        
+        return document
     }
-
-    private func modify<T: Entity>(_ query: Fluent.Query<T>) throws {
-        guard let data = query.data?.nodeObject else {
-            throw Error.noData
+    
+    /// Exequtes a Fluent.Query for an Entity
+    ///
+    ///
+    public func query<E>(_ query: RawOr<Fluent.Query<E>>) throws -> Node where E : Entity {
+        guard let query = query.wrapped else {
+            throw Error.invalidQuery
         }
-
-
-        let aqt = try query.makeAQT()
-        let mkq = MKQuery(aqt: aqt)
-
-        var document: Document = [:]
-
-        for (key, val) in data {
-            if key == idKey {
-                continue
+        
+        let collection = self[E.entity]
+        let filter = try makeQuery(query.filters, method: .and)
+        let document = try makeDocument(query.data)
+        let sort = makeSort(query.sorts)
+        let (limit, skip) = try makeLimits(query.limits)
+        
+        switch query.action {
+        case .create:
+            return try collection.insert(document).makeNode()
+        case .fetch(_):
+            // TODO: Support ComputedProperties
```

Comment:
We should throw `unsupported` error if the computed properties array is not empty